test/string_extension_test.dart

```diff
@@ -69,8 +69,20 @@ void main() {
     const messageWithMultipleLinksMarkdowned = "is this serious https://superman.com/this-is-a-<em>superman</em>-link hello guys https://batman.com/this-is-a-<em>batman</em>-link";
 
     final result = messageWithMultipleLinksMarkdowned.unMarkdownLinks(messageWithMultipleLinksUnMarkdowned);
-
     expect(result, equals(messageWithMultipleLinksUnMarkdowned));
+
+    const messageWithTextAndOneLinkUnMarkdowned = "is this serious https://superman.com/this-is-a-*superman*-link";
+    const messageWithTextAndOneLinkMarkdowned = "is this serious https://superman.com/this-is-a-<em>superman</em>-link";
+
+    final result2 = messageWithTextAndOneLinkMarkdowned.unMarkdownLinks(messageWithTextAndOneLinkUnMarkdowned);
+    expect(result2, equals(messageWithTextAndOneLinkUnMarkdowned));
+
+    // Link is not replaced because it is not a valid link
+    const messageWithFakeLinkUnMarkdowned = "is this serious-https://superman.com/this-is-a-<em>superman</em>-link";
```

Comment:
no, IMO, it is still the link, but a little bit typo. But ok for the moment, we can make it better step by step